lib/internal/test_runner/runner.js

```diff
@@ -417,14 +424,35 @@ function run(options) {
     options = kEmptyObject;
   }
   let { testNamePatterns } = options;
-  const { concurrency, timeout, signal, files, inspectPort, watch, setup } = options;
+  const { concurrency, timeout, signal, files, inspectPort, watch, setup, shards } = options;
 
   if (files != null) {
     validateArray(files, 'options.files');
   }
   if (watch != null) {
     validateBoolean(watch, 'options.watch');
   }
+  if (shards != null) {
+    validateObject(shards, 'options.shards');
+    validateNumber(shards.total, 'options.shards.total');
+    validateNumber(shards.index, 'options.shards.index');
+
+    if (shards.total <= 0) {
+      throw new ERR_INVALID_ARG_VALUE('options.shards.total', shards.total, 'total shards must be greater than 0');
+    }
+    if (shards.index <= 0) {
+      throw new ERR_INVALID_ARG_VALUE('options.shards.index', shards.index, 'shard index must be greater than 0');
+    }
+    if (shards.total < shards.index) {
+      throw new ERR_INVALID_ARG_VALUE(
+        'options.shards.index', shards.index, 'shard index must be less than total shards',
+      );
+    }
```

Comment:
```suggestion
    validateNumber(shards.total, 'options.shards.total', 0);
    validateNumber(shards.index, 'options.shards.index', 0, shards.total);
```

Nitpic: `validateNumber` accepts `min` and `max`

Reply:
thanks, done

Reply:
for the `options.shards.index` I did manual validation for better error description